contracts/batcher/TwoWayBatcher.sol

```diff
@@ -0,0 +1,85 @@
+//SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.13;
+
+import "hardhat/console.sol";
+import "@equilibria/root/number/types/UFixed18.sol";
+import "@equilibria/root/token/types/Token18.sol";
+import "@equilibria/root/token/types/Token6.sol";
+import "@equilibria/root/control/unstructured/UReentrancyGuard.sol";
+import "./Batcher.sol";
+
+contract TwoWayBatcher is UReentrancyGuard, Batcher {
+    using UFixed18Lib for UFixed18;
+    using Token18Lib for Token18;
+    using Token6Lib for Token6;
+
+    event USDCLoaned(address indexed depositor, UFixed18 amount);
+    event USDCRepaid(address indexed depositor, UFixed18 amount);
+
+    error TwoWayBatcherInvalidUSDCAmount(UFixed18 amount);
+
+    UFixed18 public usdcLoansOutstanding;
```

Comment:
same comment as verbiage -- `totalDeposited / totalSupplied`?

for these two they should have the same pattern though for example:

`deposited[account] / totalDeposited` or
`balances[account] / totalBalance`

Reply:
if we wanted to be __super__ lazy we could actually make this an `ERC20` and then this would just be the `totalSupply / balanceOf`, repay / would use mint / burn, would also let depositors more easily track their balances and transfer them around between wallets.

just an idea, might be overkill, but might clean up some required code here 🤔

Reply:
The only issue I see by making this transferrable is "stuck" USDC if they transfer the deposit token to 0 or somewhere unrecoverable. Making it non-transferrable gives a little extra protection, but I guess that's on the user if their deposit gets stuck.

I'm going to implement it as an ERC20 in a separate commit so we can see the changeset and decide if it's worth it.